### DISCUSSION TITLE
Create Multi Agent System to run on Farama Highway Env/CARLA

### DISCUSSION THREAD

**Discussion by HasarinduPerera** [FLAMEGPU/FLAMEGPU2#1134] [Q&A]
I am creating a multi agent system to experiment with an autonomous driving agent behavior as a learning project. Is it possible to create a pipeline to update the model environment as per the simulation environment (Highway/CARLA) and agent actions with Highway Env or CARLA environment. As far as I understood, I need to create some host functions to update my model environment as per the simulation environment data. Is it correct? If so, how should I approach it?

Thanks in advance!

**Comment by mondus** [comment]
Thanks for getting in touch @HasarinduPerera. We are not overly familiar with CARLA but noticed that it has some [SUMO integration](https://carla.readthedocs.io/en/latest/adv_sumo/). It looks like way SUMO is integrated is to use a client server mechanism over TCP to allow the simulations to interact. This would seem like the best way to undertake this. You could use FLAME GPU step functions to gather data for a client server architecture to communicate. I would suggest to run any server in its own thread and perhaps use the same SUMO formats.

**Reply by HasarinduPerera** [reply]
Thank you very much for your response. I'll try that and update about the progress.
